CMakeLists.txt

```diff
@@ -363,86 +373,58 @@ include_directories("${CMAKE_SOURCE_DIR}/include")
-
-  # Win64 specific rocksdb
-  set(ROCKSDB_STATIC_LIBRARY "${WINDOWS_DEP_DIR}/rocksdb/local/lib/rocksdblib.lib")
-  set(ROCKSDB_SNAPPY_LIBRARY "${WINDOWS_DEP_DIR}/snappy-msvc/tools/snappy-windows-1.1.1.8/native/snappy64.lib")
-
-  # Win64 specific doxygen
-  set(DOXYGEN_EXECUTABLE "${WINDOWS_DEP_DIR}/doxygen/tools/doxygen.exe")
-
-  # Win64 specific boost
-  set(boost_filesystem_library "${BOOST_DEP_DIR}/lib64-msvc-14.0/libboost_filesystem-vc140-mt-1_59.lib")
-  set(boost_system_library "${BOOST_DEP_DIR}/lib64-msvc-14.0/libboost_system-vc140-mt-1_59.lib")
-  set(boost_regex_library "${BOOST_DEP_DIR}/lib64-msvc-14.0/libboost_regex-vc140-mt-1_59.lib")
-
   # Win64 specific gflags
   set(BUILD_SHARED_LIBS ON CACHE BOOL "Build shared gtest library" FORCE)
-  set(gflags_library "${WINDOWS_DEP_DIR}/gflags-dev/local/lib/gflags_static.lib")
 
-  # Win64 specific glog
-  set(glog_library "${WINDOWS_DEP_DIR}/glog/local/lib/glog.lib")
+  include_directories("$ENV{HOMEDRIVE}/local/boost-msvc14/boost")
+  include_directories("$ENV{HOMEDRIVE}/local/boost-msvc14")
+  #include_directories("${WINDOWS_DEP_DIR}/*/local/include")
+  file(GLOB INCLUDE_DIRS LIST_DIRECTORIES true "${WINDOWS_DEP_DIR}/*/*/include")
+  include_directories(${INCLUDE_DIRS})
 
-  # Win64 specific cpp-netlib library
-  set(cppnetlib-uri_library "${WINDOWS_DEP_DIR}/cpp-netlib/local/lib/cppnetlib-uri.lib")
-  set(cppnetlib-client-connections_library "${WINDOWS_DEP_DIR}/cpp-netlib/local/lib/cppnetlib-client-connections.lib")
-
-
-  include_directories("${THRIFT_INCLUDE_DIR}")
-  include_directories("${BOOST_DEP_DIR}")
-  include_directories("${BOOST_DEP_DIR}/boost")
-  include_directories("${WINDOWS_DEP_DIR}/glog/local/include")
-  include_directories("${WINDOWS_DEP_DIR}/gflags-dev/local/include")
-
-  # TODO(FIXME): temporary for now
-  include_directories("${WINDOWS_DEP_DIR}/cpp-netlib/local/include")
+  ## TODO(FIXME): temporary for now
   include_directories("${WINDOWS_DEP_DIR}/linenoise-ng/local/include/linenoise")
   include_directories("${WINDOWS_DEP_DIR}/snappy-msvc/tools/snappy-windows-1.1.1.8/include")
-  include_directories("${WINDOWS_DEP_DIR}/rocksdb/local/include")
-
-  # TODO(#1990): We probably need to add our third-party link directories here
-  link_directories("${BOOST_DEP_DIR}/lib64-msvc-14.0")
-  link_directories("${WINDOWS_DEP_DIR}/linenoise-ng/local/lib")
 else()
   set(PROCESS_FAMILY "posix")
   set(MKDIR_OPTS "-p")
 endif()
 
+if(WINDOWS)
+  foreach(DEP_DIR ${WINDOWS_DEP_LIST})
+    file(GLOB_RECURSE WINDOWS_LOCAL_LIB_DIRS LIST_DIRECTORIES false "${DEP_DIR}/*.lib")
+    foreach(LIB_FILE ${WINDOWS_LOCAL_LIB_DIRS})
+      get_filename_component(LIB_DIR ${LIB_FILE} DIRECTORY)
+      list(APPEND OS_LIB_DIRS ${LIB_DIR})
+    endforeach()
+  endforeach()
+  list(REMOVE_DUPLICATES OS_LIB_DIRS)
+else()
+  set(OS_LIB_DIRS
+    "/lib"
+    "/lib64"
+    "/usr/lib"
+    "/usr/lib64"
+    "/usr/lib/x86_64-linux-gnu/"
+    )
+endif()
+link_directories(${OS_LIB_DIRS})
```

Comment:
Ah, this wont work outright since the directories in the list are not applicable to all platforms. This will generate warnings when the directories do not exists. An the `link_directories` macro will assign all as `-L` for clang/gcc.